.github/workflows/ci.yml

```diff
@@ -9,6 +9,9 @@ on:
   workflow_dispatch:
 
 env:
+  dists-artifact-name: python-package-distributions
+  sdist-artifact-name-wildcard: environ_config-*.tar.gz
```

Comment:
In fact, I tend to have a dedicated job for initial setup and that job calculates things like versions and expected filenames properly, without faking — having a full Git clone in just one job is not a big deal but this also allows to test the build closer to what happens IRL. Plus, this allows me to use the pre-setup job outputs (not just an env var) across the workflow and I indeed make these more strict with various assertions in many places.
This approach also facilitates the flow where the dists are first created and then installed for testing instead of installing from source (in addition to running the test automation from sdist). With this, it is also possible to publish the dists that were installed for testing, rather than building again.

I didn't want this patch to include too many things which is why I didn't include it initially — atomic PRs are easier to grasp.